src/Components/RequestGeneratorBuilder.js

```diff
@@ -79,13 +79,13 @@ export default class RequestGeneratorBuilder extends preact.Component {
     }
 
     resetInitialConditions = async () => {
-        await initializeFields(this.state.request.id_data).then(res => {
-            this.setState(prev => {
-                prev.request.id_data = res.new_fields;
-                prev.request.signature = res.signature;
+        await initializeFields(this.state.request.id_data).then((res) => {
+            this.setState((prev) => {
+                if (res.new_fields) prev.request.id_data = res.new_fields;
+                if (res.signature) prev.request.signature = res.signature;
 
-                const name = res.new_fields.filter(i => i.type === 'name')[0];
-                const address = res.new_fields.filter(i => i.type === 'address')[0];
+                const name = res.new_fields.filter((i) => i.type === 'name')[0];
+                const address = res.new_fields.filter((i) => i.type === 'address')[0];
```

Comment:
Why not include those in the if as well? If there is no `res.new_fields` this leads to problems as well.

Reply:
I'm reasonably certain that `SavedIdData.mergeFields()` will always return an array but sure, we might as well.